File: binary_search/programmers/practice_kit_1.py
def solution(n, times):
    answer = 0
    start, end = 1, max(times) * n # 최소값과 최대값
    mid = (start + end) // 2

    while start <= end:
        mid = (start + end) // 2

        count = 0
        for time in times:
            # 총 실행시간을 각 심사 시간으로 나눴을 때 입국심사 할수 있는 사람 수
            count += mid // time
            if count >= n: break

        if count >= n:
            answer = mid
            end = mid - 1
        else:
            start = mid + 1

    return answer

File: binary_search/programmers/practice_kit_1.java
package binary_search.programmers;
import java.util.*;
/**
 * practice_kit_1
 */
public class practice_kit_1 {

    public static void main(String[] args) {
        int n = 6;
        int times[] = {7, 10};
        Solution solu = new Solution();
        System.out.println(solu.solution(n, times));
    }

    static class Solution {
        public long solution(int n, int[] times) {
            long answer = 0;
            long start = 1;

            Arrays.sort(times);
            // 캐스팅 유무로 시간 차이 많이 나나 봄..
            long end = (long) n * times[times.length - 1];
            long mid, count;

            while (start <= end) {
                mid = (start + end) / 2;
                count = 0;

                for (int time: times) {
                    count += mid / time;
                    if (count >= n) break;
                }
    
                if (count < n) {
                    start = mid + 1;
                } else {
                    answer = mid;
                    end = mid - 1;
                }
    
            }
            
            return answer;
        }
    }
}